test/utils_test.rb

```diff
@@ -1,6 +1,49 @@
 require File.expand_path(File.join(File.dirname(__FILE__), "test_helper"))
 
 class UtilsTest < Minitest::Test
+  describe ".parse_duration" do
+    DURATIONS_FROM_EPOCH = {
+      # Basic formats
+      "P1Y1M1D"        => "1971-02-02T00:00:00.000Z",
+      "PT1H1M1S"       => "1970-01-01T01:01:01.000Z",
+      "P1W"            => "1970-01-08T00:00:00.000Z",
+      "P1Y1M1DT1H1M1S" => "1971-02-02T01:01:01.000Z",
+
+      # Negative duration
+      "-P1Y1M1DT1H1M1S" => "1968-11-29T22:58:59.000Z",
```

Comment:
No issue support negative duration, but it's not clear what the entire duration being negative would mean in the context of `cacheDuration`

Also, are negative values supported for each unit, e.g. `P1Y-1M` == 11 months?

Reply:
I'm not certain that negative durations are part of the spec, as it mentions them only once, to say:

> **2.1.6**
> duration
> non-negative quantity attributed to a time interval

From conversations in the moment.js repo ([1](https://github.com/moment/moment/issues/2408), [2](https://github.com/moment/moment/issues/3960)) it appears that this is an area where implementations vary, some supporting prefix signs, some supporting infix.

For the purposes of this PR, I'm just trying to cover the existing code, which supports a prefix sign alone.

Reply:
Some info about the xsd:duration type: http://www.datypic.com/sc/xsd/t-xsd_duration.html

It seems infix should not be valid

Reply:
Ah, cool, thanks for the extra info!

As a result of the earlier conversations I've permitted decimal hours and minutes; would you like me to back that change out, since the xsd:duration spec is tighter than the ISO one?

Reply:
Yes please

Reply:
No worries, glad to have it resolved! I've updated the branch and referred to the xsd:duration docs in the commit message.